{{cookiecutter.project_slug}}/Dockerfile__poetry

```diff
@@ -0,0 +1,51 @@
+# Modified from https://github.com/michael0liver/python-poetry-docker-example/blob/master/docker/Dockerfile
+# Provided under the MIT license: https://github.com/michael0liver/python-poetry-docker-example/blob/master/LICENCE
+
+ARG PYTHON_IMAGE_TAG=3.7-stretch
+
+# Creating a python base with shared environment variables
+FROM python:${PYTHON_IMAGE_TAG} as python-base
+
+ENV PYTHONUNBUFFERED=1 \
+    PYTHONDONTWRITEBYTECODE=1 \
+    PIP_NO_CACHE_DIR=off \
+    PIP_DISABLE_PIP_VERSION_CHECK=on \
+    PIP_DEFAULT_TIMEOUT=100 \
+    POETRY_HOME="/opt/poetry" \
+    POETRY_VIRTUALENVS_IN_PROJECT=true \
+    POETRY_NO_INTERACTION=1 \
+    PYSETUP_PATH="/opt/pysetup" \
+    VENV_PATH="/opt/pysetup/.venv"
+
+ENV PATH="$POETRY_HOME/bin:$VENV_PATH/bin:$PATH"
+
+
+# builder-base is used to build dependencies
+FROM python-base as builder-base
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+        curl \
+        build-essential
+
+# Install Poetry - respects $POETRY_VERSION & $POETRY_HOME
+ENV POETRY_VERSION=1.0.5
+RUN curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python
```

Comment:
Maybe a pip install would be better here? I'm not really a fan of `curl $url | python`. Something like [this SO answer](https://stackoverflow.com/a/54763270) maybe? This would also cut down on the fat, the Dockerfile is quite complex right now and we don't want to overwhelm the user with implementation details.

A [multi-stage build](https://stackoverflow.com/a/57886655) would also be pretty awesome, but it would increase complexity again. Not sure which I'd prefer...

Reply:
Sure, happy to change this to a pip install. 

This Dockerfile is actually already multi-stage, employing a similar method to the linked stackoverflow answer. The main difference is that this Dockerfile uses poetry to create a venv, whereas the stackoverflow answer uses poetry to create a requirements.txt and then uses pip to create the venv. Since it's not in the final stage of the build (so in both cases poetry is not included in the final image), I don't think it makes much difference.

Reply:
I see, thanks for the explanation